src/integer_mod_q/mat_zq/set.rs

```diff
@@ -0,0 +1,204 @@
+use crate::utils::coordinate::evaluate_coordinates;
+use flint_sys::fmpz_mod_mat::fmpz_mod_mat_set_entry;
+use std::fmt::Display;
+
+impl SetEntry<&Z> for MatZq {
+    /// Sets the value of a specific matrix entry according to a given `value` of type [`Z`].
+    ///
+    /// Parameters:
+    /// - `row`: specifies the row in which the entry is located
+    /// - `column`: specifies the column in which the entry is located
+    /// - `value`: specifies the value to which the entry is set
+    ///
+    /// # Example
+    /// ```
+    /// use math::integer_mod_q::MatZq;
+    /// use crate::math::traits::SetEntry;
+    /// use math::integer::Z;
+    /// use std::str::FromStr;
+    ///
+    /// let mut matrix = MatZq::new(5, 10, 7).unwrap();
+    /// let value = Z::from(5);
+    /// matrix.set_entry(1, 1, &value).unwrap();
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of type [`MathError::OutOfBounds`]
+    /// if the number of rows or columns is greater than the matrix or negative.
+    fn set_entry(
+        &mut self,
+        row: impl TryInto<i64> + Display + Copy,
+        column: impl TryInto<i64> + Display + Copy,
+        value: &Z,
+    ) -> Result<(), MathError> {
+        let (row_i64, column_i64) = evaluate_coordinates(self, row, column)?;
+
+        // Calculate mod q before adding the entry to the matrix.
+        let value: Zq = Zq::from_z_modulus(value, &self.get_mod()?);
+
+        unsafe {
+            // get entry and replace the pointed at value with the specified value
+            fmpz_mod_mat_set_entry(&mut self.matrix, row_i64, column_i64, &value.value.value)
+        }
+
+        Ok(())
+    }
+}
+
+impl SetEntry<&Zq> for MatZq {
+    /// Sets the value of a specific matrix entry according to a given `value` of type [`Zq`].
+    ///
+    /// Parameters:
+    /// - `row`: specifies the row in which the entry is located
+    /// - `column`: specifies the column in which the entry is located
+    /// - `value`: specifies the value to which the entry is set
+    ///
+    /// # Example
+    /// ```
+    /// use math::integer_mod_q::MatZq;
+    /// use crate::math::traits::SetEntry;
+    /// use math::integer_mod_q::Zq;
+    /// use math::integer::Z;
+    /// use std::str::FromStr;
+    ///
+    /// let mut matrix = MatZq::new(5, 10, 7).unwrap();
+    /// let value = Zq::from_str("5 mod 12").unwrap();
+    /// matrix.set_entry(1, 1, &value).unwrap();
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of type [`MathError::OutOfBounds`]
+    /// if the number of rows or columns is greater than the matrix or negative.
+    fn set_entry(
+        &mut self,
+        row: impl TryInto<i64> + Display + Copy,
+        column: impl TryInto<i64> + Display + Copy,
+        value: &Zq,
+    ) -> Result<(), MathError> {
+        let (row_i64, column_i64) = evaluate_coordinates(self, row, column)?;
+
+        // TODO error when modulus not equal
```

Comment:
Why is this TODO open?

Reply:
The error exists in another PR. It will be added when this PR is merged into dev.